test/foundry/GeometricPriceBook/integration/Coefficients.t.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: -
+// License: https://license.clober.io/LICENSE.pdf
+
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+import "../../../../contracts/markets/VolatileMarket.sol";
+import "../../../../contracts/mocks/MockQuoteToken.sol";
+import "../../../../contracts/mocks/MockBaseToken.sol";
+import "../../../../contracts/OrderNFT.sol";
+
+contract GeometricPriceBookIntegrationTest is Test {
+    MockQuoteToken quoteToken;
+    MockBaseToken baseToken;
+    OrderNFT orderToken;
+
+    function setUp() public {
+        quoteToken = new MockQuoteToken();
+        baseToken = new MockBaseToken();
+        orderToken = new OrderNFT(address(this), address(this));
+    }
+
+    function testCoefficients(uint128 a, uint128 r) public {
+        if ((uint256(r) * a) / 10**18 <= a) {
+            vm.expectRevert();
+            new VolatileMarket(
+                address(orderToken),
+                address(quoteToken),
+                address(baseToken),
+                1,
+                0,
+                0,
+                address(this),
+                a,
+                r
+            );
+        }
+    }
+
+    function _testCoefficients(uint128 a, uint128 r) internal {
```

Comment:
Fuzzing test for this function should be added.

Reply:
Resolved.